### DISCUSSION TITLE
Cannot get traceId in SpringCloudGateway filters

### DISCUSSION THREAD

**Discussion by chilejiang1024** [apache/skywalking#9125] [Q&A] [answered]
skywalking oap version: 9.0.0
skywalking agent version: 8.10.0
spring cloud gateway version: 3.0.3

Already put the optional plugins (gateway-3.x-plugin, spring-webflux-5.x-plugin) into agent's plugins floder.
Is there anyway to get traceId from custom filters?
I found that trace context has been created in `beforeMethod` of `org.apache.skywalking.apm.plugin.spring.cloud.gateway.v3x.NettyRoutingFilterInterceptor`, so,  filters before `NettyRoutingFilter` cannot get traceId. 
I tried to add an interceptor to intercept my filter (with the highest order), but if I doing this, traceId never changed...
Or, can I create trace context in a higher order filter, so that traceId can be used in lower custom filters.
Please help, Thanks.

**Comment by feixiangshenhua** [comment]
I have the same question,do you solve this problem now?

**Comment by wu-sheng** [comment, marked as answer]
I think this should help, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-webflux/ as spring gateway is a kind of webflux.

**Reply by feixiangshenhua** [reply]
yeah，I practice， it's  effective。thanks so much。
